moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1182,8 +1187,13 @@ void TrajectoryExecutionManager::stopExecution(bool auto_clear)
   else if (execution_thread_)  // just in case we have some thread waiting to be joined from some point in the past, we
                                // join it now
   {
-    execution_thread_->join();
-    execution_thread_.reset();
+    execution_thread_mutex_.lock();
```

Comment:
We could remove this else if with just else statement. What do you think?

Reply:
Yes, that makes sense.

Reply:
This was an instance of [double-checked locking](https://en.wikipedia.org/wiki/Double-checked_locking).
It might not be necessary here, but just removing it because the read is outside the mutex is no good argument.

Reply:
It could not make any harm, with else statement and additional check, the same thing is achieved. Isn't it? Fix me if I am wrong.

Reply:
Please look up the concept.
Double-checked locking is used when the time used to lock the mutex is relevant and the mutex is usually not needed (the latter clearly being the case here).

Reply:
@livanov93: I pushed a cleanup commit, to restore the double-checked locking.